src/attestation/Attestation.ts

```diff
@@ -15,12 +15,71 @@ import IRequestForAttestation from '../types/RequestForAttestation'
 import TxStatus from '../blockchain/TxStatus'
 import { factory } from '../config/ConfigLog'
 import Identity from '../identity/Identity'
-import IAttestation from '../types/Attestation'
+import IAttestation, { CompressedAttestation } from '../types/Attestation'
 import { revoke, query, store } from './Attestation.chain'
 import IPublicIdentity from '../types/PublicIdentity'
 
 const log = factory.getLogger('Attestation')
 
+function attestationErrorCheck(attestation: IAttestation): void {
```

Comment:
I took the check from inside the `Attestation` constructor and moved it as a separate function. 

As I used it to check inside the `compressAttestation` and the constructor.